executor/executor.go

```diff
@@ -170,7 +170,9 @@ func (e *Executor) Run(keys state.Keys, f func() error) {
 			if v == state.Read {
 				// If we don't need exclusive access to a key, just mark
 				// that we are reading it and that we are a reader of it.
-				t.reading = append(t.reading, lt)
+				// TODO: currently a no-op when we add ourself multiple times
```

Comment:
I'm not sure this is that bad and/or deserves a TODO.

The alternative might be trying to track this across keys but that'll get complex with locking imo.